packages/repository-json-schema/src/__tests__/unit/filter-json-schema.unit.ts

```diff
@@ -157,15 +171,52 @@ describe('getFilterJsonSchemaFor', () => {
   }
 });
 
-describe('getFilterJsonSchemaForOptions', () => {
+describe('getFilterJsonSchemaForOptionsSetTitle', () => {
+  let customerFilterSchema: JsonSchema;
+
+  beforeEach(() => {
+    customerFilterSchema = getFilterJsonSchemaFor(Customer, {setTitle: true});
+  });
+
   it('returns "title" when a single option "setTitle" is set', () => {
-    const orderFilterSchema = getFilterJsonSchemaFor(Order, {setTitle: true});
-    expect(orderFilterSchema).to.have.key('title');
+    expect(customerFilterSchema.title).to.equal('Customer.Filter');
+  });
+
+  it('returns "include.title" when a single option "setTitle" is set', () => {
+    const filterInclude = customerFilterSchema.properties!
+      .include as JsonSchema;
+    expect(filterInclude.title).to.equal('Customer.IncludeFilter');
+  });
+
+  it('returns "scope.title" when a single option "setTitle" is set', () => {
+    const filterInclude = customerFilterSchema.properties!.include;
+    const filterScope = ((filterInclude as JsonSchema).items as JsonSchema)
+      .properties!.scope as JsonSchema;
+    expect(filterScope.title).to.equal('Customer.ScopeFilter');
+  });
```

Comment:
Is there a better way than as ... as ... ?

Reply:
I think the problem is in the way how `@types/json-schema` describes the schema object. Each property definition can be either `JSONSchema6` (what we use) or `boolean`. TypeScript cannot tell that we are always using `JSONSchema6` and never `boolean`, that's why you have to use explicit `as` casts.

It may be possible to work around this problem by explaining TypeScript that `getFilterJsonSchemaFor` is returning not just some random `JsonSchema`, but a schema describing an object with few well-known properties (`where`, `fields`, etc.) that are defined using a `JSONSchema6` object, not as a `boolean` value. There may be a way how to do that in a generic way, something along the way of "for every key of `Filter` type, tell TypeScript that the schema describes a property of that name.

I am not sure if it's worth the effort though.

To simplify your tests, maybe you can use deep property checks, e.g. using [propertyByPath](http://shouldjs.github.io/#assertion-propertybypath) assertion. An mock-up to illustrate what I mean:

```ts
expect(customerFilterSchema.properties)
  .to.have.propertyByPath(
    ['include', 'items', 'properties', 'scope', 'title']
  )
  .equal('Customer.ScopeFilter');
```